File: leetCode.com/typescript/linkedList/removeNthNodeFromEndOfList.ts
// Given the head of a linked list, remove the nth node from the end of the list and return its head.

function removeNthFromEnd(head: ListNode | null, n: number): ListNode | null {
    const preHead = new ListNode(0, head)
    let fast = preHead
    let slow = preHead
    let listLength = 0

    while (fast.next) {
        fast = fast.next
        listLength++
    }
    for (let i = 0; i < listLength - n; i++){
        if (slow.next)
            slow = slow.next
    }
    slow.next = slow.next ? slow.next.next : null
    return preHead.next
};

File: leetCode.com/python/linkedList/removeNthNodeFromEndOfList.py
# Given the head of a linked list, remove the nth node from the end of the list and return its head.

from typing import Optional
from classes import ListNode

def removeNthFromEnd(head: Optional[ListNode], n: int) -> Optional[ListNode]:
    preHead = ListNode(0, head)
    fast = preHead
    slow = preHead
    listLength = 0
    while(fast.next):
        fast = fast.next
        listLength += 1
    for i in range(0, listLength-n):
        slow = slow.next
    slow.next = slow.next.next
    return preHead.next